vars/mtaBuild.groovy

```diff
@@ -28,6 +28,21 @@ def call(Map parameters = [:]) {
                                       parameters, parameterKeys,
                                       stepConfiguration, stepConfigurationKeys, stepDefaults)
 
+        JAVA_HOME_CHECK : {
```

Comment:
The version of Java must be checked also if it is in the PATH.

Reply:
The target of that pull request is basically to move the validation from the pipeline into the steps. Before, having the validation in the pipeline, there was no validation when java was simply in the path. Actually this resulted in a validation failure, which in turn failed the build.

In the step it was possible to work also when java was only in the path. In the step there was no validation at all.

The approach here with only checking the java version when java is in the path is backward compatible with the old approach. In case somebody used the step before outside a pipeline performing a tool validation there was no exception caused by a version failure. From that point of view we are fully backwardcompatible regarding the behaviour. In case our version check is reasonable, there should - of course - be a failure when the tool is invoked.

toolValidate is not able to work with the path to the executable (including the executable itself. toolValidate works with the ```JAVA_HOME``` and concatenates ```bin/java```. It is also not possible in a reliable way to calculate JAVA_HOME from the executable path. Stripping the trailing ```bin/java``` does not work since we may have a symblic link somewhere in the path. As long as the link would be in ```/usr/bin``` is would work (despite the fact that ```/usr``` is not a valid ```JAVA_HOME```.  Having a sybolic link to the java executable somewhere else in the path may fail ...   Anyway stripping down is fishy ...

I suggest to fix the missing validation when the tool is in the path in another PR.

By the way: imagine we have ```JAVA_HOME``` set, but another java executable in the path. That is a smell, but it can happen. What do we validate in this case? Would be reasonable to validate the version from the path.

Reply:
I created issue #102 for changing the validation accordingly (```HOME``` and executable from path).